executor/join.go

```diff
@@ -665,46 +651,99 @@ func (e *HashJoinExec) handleFetchInnerAndBuildHashTablePanic(r interface{}) {
 }
 
 func (e *HashJoinExec) fetchInnerAndBuildHashTable(ctx context.Context) {
-	// innerResultCh transfers inner chunk from inner fetch to build hash table.
-	innerResultCh := make(chan *chunk.Chunk, 1)
-	doneCh := make(chan struct{})
-	// TODO: just used in test now, thus we return directly in this if-branch.
-	// Remained work:
-	// 1. parallel build hash tables for the inner sub-partitions
-	// 2. partition the outer relation into sub-partitions
-	// 3. parallel join corresponded inner-partitions with outer-partitions
+	// Step1: fetch all the data of the inner relation into memory.
+	if err := e.fetchInnerRows(ctx); err != nil {
+		e.innerFinished <- err
+		return
+	}
+	if e.finished.Load().(bool) {
+		return
+	}
+
+	// Step2: do parallel partition if needed.
+	needPartition := false
```

Comment:
Should it be set as a member variable of `HashJoinExec`?

Reply:
Noop, we check whether to use radix-hashjoin using 2 steps.
First check the variable enable_radix_join, suppose it true.
Next we check:
1. whether the data in the small-table is skewed heavily, if so, we do not introduce the partition phase.
2. whether the size of the small-table can fit into L2-cache, if so, we do not introduce the partition phase.

Reply:
> whether the size of the small-table can fit into L2-cache, if so, we do not introduce the partition phase.

the size of the whole small table?

Reply:
`small table + big table + hash table`

It seems that we can estimate this during plan building?

Reply:
Yeah, I also hope to let the planner to estimate this and build cost models for NoPartitionedHashJoin and RadixHashJoin separately, but considered the estimation error, the estimation maybe is not that correct.

Reply:
OK, but these are two different tasks.
I'll extract it as `RadixHashJoinExec`.